### ISSUE TITLE
Is this project dead?

### ISSUE THREAD

**Issue by Tidle999** [FlauraPlantPot/Flaura#4]
This project is awesome. But it seems to be not continued anymore, right? There are no activities for months now. I just want to now before ordering all the stuff. 
Using mqtt would be a great option in my opinion to integrate in smart home installations. 

Regards

**Comment by MartinVinter** [comment]
Not sure. Last activity was in december, which isn't that long ago. On the other hand, in his last the last comments around Blynk  made by the auther on thingiverse he sounded frustrated (and rightly so).

I too think this is a super-cool project. I absolutely love it. I didn't notice that there wasn't any working code for the thing, so now I have a pile of parts that need the code to go with them. They weren't cheap either.

I'd like to see if I can get the code to work without app-support. I seems to me it should be possible to get it to work with just a potentiometer for setting the moisture threshold and the other parameters hardcoded (accessible with a pc via usb). App-control would be super-cool but it would require more from me than I'm willing to put into it, in part because my skills just doesn't go that far.

**Comment by Tidle999** [comment]
If you could do that that would be awesome. 😊

**Comment by pypypeanuts** [comment]
Love this project. I'll give this a try and eventually customize it with my NodeMCU.
MQTT would be really nice.
Possibly another sensor would be useful to alert if there is insufficient water in the tank.

**Comment by MartinVinter** [comment]
It already has a sensor for detecting water level (a bunch of wires held at intervals in the tank. Contact = water).

A NodeMCU won't fit in the print. You need an ESP32. The ones he use in the project are a cheap knockoff that often don't even work, but dollarTek has made one that fits and has the pins in the same places.

I have managed to clean the code of all the Blynk stuff and can uploade it without errors, but it doesn't quite work. The pump just keeps running. I'd love it if anyone was willing to help out. I am not a great coder by any standards, and especially not in Arduino.

**Comment by pandel** [comment]
@MartinVinter - the best solution would be, you fork the main repo and push your changes to your fork. Then everyone can clearly see what you modified and perhaps finally fix the error with the pump - I am interested in you changes and may be I can help with the code, but have not started building a plant pot yet even though I have all parts at hand.

(Forget about pull requests :-), he does not seem to react to anything that is happening here)
https://docs.github.com/en/get-started/quickstart/contributing-to-projects

**Comment by MartinVinter** [comment]
@pandel : Since I wrote my last message, I came up with an entirely different, more radical and much easier way of getting it to work.
I think, that with ESPHome (see https://esphome.io/) and Home assistant (see https://www.home-assistant.io), it is possible to get this to work while also getting the functionality that was previously handled by blynk. It's a much better option of the non-networked version I was going for before.

The best thing is that ESPHome takes almost all the coding out of setting up a device like this. All you have to do is to configure (mostly copy paste, actually) some yaml-files and you're rolling. I think it may use more battery though, but who cares. 

My Home Assistant instance needs some upgrading before I can get cracking, but it wont take long, I think. With a bit of luck I'll have something in a few weeks.

**Comment by Tidle999** [comment]
Sounds great. You mean that you send data to home assistant to trigger the water pump with HA?

**Comment by MartinVinter** [comment]
Yes. The flow would be like this, I imagine:

Wakeup and measure, send to HA.
HA triggers water if needed ---> sleep one hour, measure again. Repeat.
Sleep for 24 hours.

**Comment by Tidle999** [comment]
Sound good. Sleeping for 24h only when the last measurement was fine and no water needed,right?

**Comment by MartinVinter** [comment]
Are any of you who commented here handy with 3d designing tools? 
The thing is, that I tried assembling Flaura, and it turns out that its not at all as easy as advertised by the creator to get water-tight. It is possible, but it also takes some effort and testing.

It is designed with many joints that span pretty big surfaces and with a lot of nooks and crannies that make for a lot of places water can seep out. I think it should be possible to design to the thing much simpler. 

Are any of you up for that?

**Comment by ghost** [comment]
Does somebody have the programming skills to do a very simplified fork (no app), like this:

- [ ] Every 12h measure soil moisture
- [ ] If moisture is below 50%, run pump until it is 80%
- [ ] If tank is empty, don't run pump (maybe add an indicator LED anywhere)

If we have this, we could expand with something like this:

- [ ] Local website on ESP32 to show history of watering
- [ ] Settings page
- [ ] ...

**Comment by MartinVinter** [comment]
I still think it is possible to get all of the above working with ESPHome and HomeAssistant. I've been messing with it the past few weeks, and it looks very promising. https://esphome.io/index.html It would also give you the two nice2's you mention.

I've also tried to print and assemble the pot itself, and noted that it is very hard to get a proper seal. You have to make 5 seals that are watertight, and 3 of them are tubes. If you modify the design slightly, you could print the whole thing in one piece with pipe connectors where hoses cross walls in stead of drilled holes that has to be sealed. It's takes much less assembly, in that way.
However, the source files are made in Inventor, which is expensive as heck. If we could find someone who has Inventor or the paid version of Fusion, the files could be converted into something I could work with for. Does anyone know someone who has that?A

**Comment by Tidle999** [comment]
i think i have the possibility to use it at work. i think i can simply import and export, but i dont have any idea how to use this software for design changes.

**Comment by MartinVinter** [comment]
That sounds great. Then all you have to do is to download the IPT-files from Thingiverse and save them in the Fusion 360 format.

**Comment by ghost** [comment]
I will have a look at ESPHome.io. It sounds promising.

**Comment by Tidle999** [comment]
hera are the exported step files, can you use them? i was not able to export in a specific fusion 360 file format.
https://easyupload.io/m/92ussh

**Comment by ghost** [comment]
I was able to export the files in .f3d-Format. You can find them here:
[https://easyupload.io/m/pcc52v](https://easyupload.io/m/pcc52v)

**Comment by MartinVinter** [comment]
Thank you crabucate and Tidle999, it works. I've downloaded all the files and checked that I can open and edit them.

I've also created an invite on the ESPHome forum introducing the project and asking anyone who wants to to join and chip in with some coding.

I'll start editing the 3d models. The idea is this:
-The idea is to join the inner, the outer parts and the bottom in one print. Hoses will cross walls with pipe connectors that are part of the print to ensure a tight seal and easier assembly. 
-The watering pipe will go through the soil instead of the tank because the tank will be a (mostly) inaccessible cavity in the new print.
-The water level sensor will be replaced by an "almost empty"-sensor. It is made by driving two galvanized screws from the bottom into the tank and using them as electrodes (contact = water present). It is easier to get a seal and they will also not corode.
The downside is this: It will be nessesary to print the lower part of the outside shell separately because the bottom where all the electronics sit will require a lot of support material. If the outside wall is present, the support material can't come off. It will make a visible joint between the lower part of the outside and the rest of the pot, which will make the finished pot less pretty.

If you have any suggestions, I'd love to hear them :-)

**Comment by Tidle999** [comment]
Are there any news here?

**Comment by MartinVinter** [comment]
There is. Lots, actually. But the conversation has mostly continued on ESPHome's forum: https://community.home-assistant.io/t/invitation-project-suggestion-an-automated-potted-plant/431846/38

Join us over there. You'll also find a link to the new git-project there with the STL's for the 3d print and the code for ESPhome (doesn't work yet though)

**Comment by MikeBailleul** [comment]
> Are there any news here?

I've worked on the project, redesigned the PCB, 3D models and code, it now works on Home Assistant with ESPHome, try it [here](https://github.com/MikeBailleul/plantidote-smart-flower-pot) ;)

<img width="843" alt="" src="https://github.com/FlauraPlantPot/Flaura/assets/25762591/f1bb37ea-9227-414c-9ade-7ab55bce82f1">

**Comment by MartinVinter** [comment]
No, not dead. We just moved the talking to ESPHome's forum for a while and then the dialogue sort of died of and I just worked on the project alone. Inspired by Mathgaming, I ended up with a radically different approach to the printed parts. It is not as pretty but way more practical. It also holds more water with the same footprint, I think

![image](https://github.com/FlauraPlantPot/Flaura/assets/61094818/3004da80-5591-40fd-8b45-5a0cb51a923e)

The inside container is for the plant, the outside is the resovoir. The inside part is threaded in the bottom so you can take it out and replant without messing up the resovoir.
The electronics are assembled on the green board which can be easily removed from the purple container. The flat thing on the side is a mount for a solar panel. There is a lid that snaps on as well so assembled, it looks like this:

![image](https://github.com/FlauraPlantPot/Flaura/assets/61094818/644817a9-87a5-4c9b-b00a-fa76de5202e0)

**Comment by MartinVinter** [comment]
I've printed a few, and they seem to work. My main issue is that with PLA it seems to slowly disintegrate while PETG needs a coating to become completely watertight. How do you achieve completely watertight printed parts that does not allow a single drop of water to slowly seep out between the layers here and there?

**Comment by MartinVinter** [comment]
Your planter looks really nice, btw :)

**Comment by MikeBailleul** [comment]
> I've printed a few, and they seem to work. My main issue is that with PLA it seems to slowly disintegrate while PETG needs a coating to become completely watertight. How do you achieve completely watertight printed parts that does not allow a single drop of water to slowly seep out between the layers here and there?

Have you identified a specific point of leakage? Or is it leaking kind of everywhere?

The biggest issue I had was a leak from the borders so I added a reinforcement, see pic.
![](https://github.com/FlauraPlantPot/Flaura/assets/25762591/a2aa703d-ebab-4e23-bcd2-cf242472baa8)

I also made the bottom a bit thicker and printed at infill 90% gyroid.

**Comment by MartinVinter** [comment]
It's more or less everywhere, but particularly wherever the extruder changes direction. Having rounded corners has helped. Over-extruding (to the point where the extruder is skipping), randomizing the Z-seam and increasing the temperature has as well but I still have to coat each part after printing. It's a process I could do without. Are your prints completely watertight with no need for coating? with PETG or PLA? And if so, what settings do you use?

**Comment by MikeBailleul** [comment]
Let's try to figure out what could be the issues:

1. printer printing okay? nozzle okay? No under extrusion?
2. Filament quality? I used PLA Overture 
3. Slicing issue? I print external perimeter first + 7 perimeters

I recommend you to take a look at the print settings I detailed in my README: https://github.com/MikeBailleul/plantidote-smart-flower-pot#print-settings

Maybe it could also help to download one of the .3mf files and check the settings on PrusaSlicer. 

Also, I printed in PLA mostly and it worked fine, I also tried PETG and it worked.

**Comment by MartinVinter** [comment]
1) faily new, and the printer appears fine and produces no other errors. I don't think that is it.
2) I use PETG from Eryone. It generally gets good reveiws. ...But it certainly isn't pla - so that's one difference. But PLA is known to disintegrate on prolonged contact with water. Won't that be an issue? How long has your oldest pot lasted so far?
3) That is probably it. You print with a .4 nozzle, so that makes the wall thickness 3 mm-ish...? I print with a .6 nozzle for speed and have only 3 wall layers and 2 mm thickness. According to something I read somewhere, a .6 nozzle should get me a more watertight print, but it doesn't seem to be the case. I'll try changing the nozzle to a smaller one max the wall count.

**Comment by MikeBailleul** [comment]
2. Biodegradable is a broad term and PLA being called such thing can lead to misunderstanding. I mean, it is but under specific conditions like high temperature etc. In the wild, it takes something like 80 years to decompose.
Check this video: https://www.youtube.com/watch?v=X_Gh-3PQhiE&ab_channel=HobbyHoarder
My oldest print on this project holding water is just weeks but I have already printed other vase, containing water for months showing no sign of desintegration.

3. Interesting point, I didn't think about nozzle diameter.
I remember reading that having a larger number of walls/perimeters is more important for waterproofing. I'll try to find that resource and post it here later.

EDIT: Found it: https://blog.prusa3d.com/watertight-3d-printing-pt1-vases-cups-and-other-open-models_48949/
According to this blog post, number of walls is what matters the most but thickness helps as well. I personally didn't change the thickness so it's 0.45mm by default.
Also, even though the blog post specify that PLA won't degrade, they still recommend to print with PETG if possible which I will do from now on because PLA can slightly becomes weaker even though they did not see weaknesses themself. I will see how my PLA Plantidote is in a few months...

**Comment by MartinVinter** [comment]
2: Oh, I wasn't worried about it composting but rather that it is hydrophillic and will soak water over very long periods, becoming brittle and eventually leaky after months or even years.

Your questions about filament got me thinking that it actually did become more leaky after changing brands fra Amazons stock PETG which isn't available anymore to Evyone. I bumped up the temperature by 5C, and the part that holds the plant was watertight. Another 5 and I think the parts that are harder to get tight will too. It sometimes helps to explain whatever issues one is facing to others, even if they don't actually provide the solution directly :)

I went through your design in detail, and I really like it. Does the design scale in size as well? The thing is that the plants often grow out of the small planter quite quickly, especially when you have a timer to keep them watered all the time.

What was the purpose of the PCB redesign? I can see it is different, in the layout but I can't seem to spot any fundamental changes.

In my half-finished design, the water-level prope was meant to be made with a few 2 or 3 mm bolts of varying length that were added through the bottom with Silicone added when half-way through or heated with a soldering iron once fully inserted to make a seal. The point was to get a prope that was more resistant to corrosion than the thin wires, which is important when they are glued in. I never tested that principle though. Did you?

**Comment by MartinVinter** [comment]
Oh, and one last thing, just FYI. I had issues with the MOZFET. In my experience it sometimes worked, and sometimes not. I think the issue was that the threshold for switching it was on the fringes of what the PCB/board provided, but I never fully tested it. I switched to the PSMNB3R4 and in my experience, it works better. It was a really annoying issue, so I thought I might save you the trouble if you bump into it as well.

**Comment by MikeBailleul** [comment]
Agreed, I sometimes write down my issue as if I explain it to someone and then the solution comes to me ;)

Out of the box, it will not scale, mainly because of the bottom part. But I think it should not be hard to scale only the other parts in fusion 360, I will give a try later. But there is also another issue, we will quickly reach the limit of the printing plate. In my case, my plate would allow me to scale in 30% more (which is not so bad).

To be honest, I got motivated redesigning to learn how to design a PCB :)

Here are the changes:
- I changed the value of the resistor for the MOSFET because according to my calculations, it was not enough. I had the same issues you had, sometimes triggering, sometimes not, very random. Glad to know it also works by changing the MOSFET.
- I changed the pin used for the button, it was a strapping pin (pin 0) and it caused me a lot of issues while developing the code so I changed it to another one (pin 13).
- I freed one pin and removed one resistor for reading the battery voltage because I discovered that the LolinD32 has a built-in feature for that :)
- I connected all LolinD32 pins to the board, this way, we can use other pins to add features if the project evolves more. Because of that I had to make the cutout a bit of a weird rectangular shape.
- I removed one pin for the water level, because I felt I didn't need that many

Thanks for asking, I will add them to the readme.

Regarding the water-level sensors, I remember the screws idea from the HA forum, I tried that several times but it did not work out well for me and I wanted to know the higher water levels (which would require very long screws). I think it can still be an idea to explore. Though I'm not sure about the corrosion of wires since I use them only 1 second per day. This would need more testing...
